unified_planning/engines/plan_validator.py

```diff
@@ -168,6 +181,7 @@ def _validate(
                     reason=FailedValidationReason.INAPPLICABLE_ACTION,
                     inapplicable_action=ai,
                     trace=trace,
+                    calculated_interpreted_functions=calculated_interpreted_functions,
```

Comment:
You can directly get the calculated interpreted functions here and in all the cases where you return a ValidationResult with an INVALID status
```suggestion
                    calculated_interpreted_functions=simulator.get_knowledge(),
```